server/api-manager.js

```diff
@@ -926,11 +926,35 @@ class MindbodyQueries {
     let funcc = () => {
       return request.makeRequest();
     };
-    const backoff = (retries, fn, delay = 500) => fn().catch(err => (retries > 1 && !this.atLimit()) ? pause(delay).then(() => backoff(retries - 1, fn, delay * 2)) : Promise.reject(err));
+    function pause(milliseconds) {
+      return new Promise(resolve => {setTimeout(resolve, milliseconds)});
+    };
+    const maxRetries = 5;
+    // Madison's original definition of backoff:
+    //const backoff = (retries, fn, delay = 500) => fn().catch(err => (retries > 1 && !this.atLimit()) ? pause(delay).then(() => backoff(retries - 1, fn, delay * 2)) : Promise.reject(err));
+    const backoff = (retries, fn, delay = 500) => {
+      return fn().catch((err) => {
+        if (retries == maxRetries) {
+          console.log("");
+          console.log(" --- Initial failure of request: " + request.url);
+        }
+        console.log("Error message was: " + err);
+        if (retries > 1 && !this.atLimit()) {
+          console.log("Counting down #" + retries + " retries...");
+          return pause(delay).then(() => {
+            return backoff(retries - 1, fn, delay * 2);
+          });
+        } else {
+          console.log("Request " + request.url + " was finally rejected.");
+          console.log("");
+          return Promise.reject(err);
+        }
+      });
+    }
 
     if (!this.atLimit()) {
       //return request.makeRequest();
-      return request.makeRequest()
+      return backoff(maxRetries, funcc)
```

Comment:
I just barged ahead and made this change. Before, we only did backoff when requesting the additional pages for a multipage request. I think we should use backoff all the time. As one example, Asher's "getClassVisits" requests (that trigger this bug) are not part of a multipage request, but they have need backoff to work. So, I think we should do backoff for all requests.